tests/TenantDatabaseManagerTest.php

```diff
@@ -220,6 +221,126 @@
+    ]);
+
+    Event::listen(TenantCreated::class, JobPipeline::make([CreateDatabase::class])->send(function (TenantCreated $event) {
+        return $event->tenant;
+    })->toListener());
+
+    $name = 'foo' . Str::random(8);
+    $tenant = Tenant::create([
+        'tenancy_db_name' => $name,
+        'tenancy_db_host' => 'mysql2',
+    ]);
+
+    /** @var MySQLDatabaseManager $manager */
+    $manager = $tenant->database()->manager();
+
+    $manager->setConnection('mysql2');
+    expect($manager->databaseExists($name))->toBeTrue();
+});
+
+test('database credentials can be provided to PermissionControlledMySQLDatabaseManager by specifying a connection', function () {
+    config([
+        'tenancy.database.managers.mysql' => PermissionControlledMySQLDatabaseManager::class,
+        'tenancy.database.template_tenant_connection' => 'mysql',
+        'database.connections.mysql2' => [
+            'driver' => 'mysql',
+            'host' => 'mysql2',
+            'port' => 3306,
+            'database' => 'main',
+            'username' => 'root',
+            'password' => 'password',
+            'unix_socket' => env('DB_SOCKET', ''),
+            'charset' => 'utf8mb4',
+            'collation' => 'utf8mb4_unicode_ci',
+            'prefix' => '',
+            'prefix_indexes' => true,
+            'strict' => true,
+            'engine' => null,
+            'options' => extension_loaded('pdo_mysql') ? array_filter([
+                PDO::MYSQL_ATTR_SSL_CA => env('MYSQL_ATTR_SSL_CA'),
+            ]) : [],
+        ],
+    ]);
+
+    // Create a new random database user with privileges to use with mysql2 connection
+    $username = 'dbuser' . Str::random(4);
+    $password = Str::random('8');
+    $mysql2DB = DB::connection('mysql2');
+    $mysql2DB->statement("CREATE USER `{$username}`@`%` IDENTIFIED BY '{$password}';");
+    $mysql2DB->statement("GRANT ALL PRIVILEGES ON *.* TO `{$username}`@`%` identified by '{$password}' WITH GRANT OPTION;");
+    $mysql2DB->statement("FLUSH PRIVILEGES;");
+    config(['database.connections.mysql2.username' => $username]);
+    config(['database.connections.mysql2.password' => $password]);
+
+    Event::listen(TenantCreated::class, JobPipeline::make([CreateDatabase::class])->send(function (TenantCreated $event) {
+        return $event->tenant;
+    })->toListener());
+
+    $name = 'foo' . Str::random(8);
+    $tenant = Tenant::create([
+        'tenancy_db_name' => $name,
+        'tenancy_db_connection' => 'mysql2',
+        'tenancy_db_username' => $username = 'user_for_new_db' . Str::random(4),
+        'tenancy_db_password' => Str::random(8),
+    ]);
+
+    /** @var PermissionControlledMySQLDatabaseManager $manager */
+    $manager = $tenant->database()->manager();
+
+    expect($manager->userExists($username))->toBeTrue();
+    expect($manager->databaseExists($name))->toBeTrue();
+});
+
+test('tenant database can be created by using the username and password from tenant config', function () {
```

Comment:
> What does this test exactly? What role does the username & password have here?

The main purpose of this test is to verify that we can use separate usernames and passwords through the tenant config key. Eg, I have the same host, but I want to use different credentials. 

But now looking at the test, I'm using the same username and password as the `mysql` config has. So the test is kind of useless.

Reply:
> verify that we can use separate usernames and passwords through the tenant config key

Yeah but are we actually checking anything? I just see that you're passing these credentials, but then aren't making any assertions about them.

Reply:
>Yeah but are we actually checking anything? I just see that you're passing these credentials, but then aren't making any assertions about them.

I'm asserting that the database is created. That means we are using the given credentials.

Reply:
The name is: "can be created **by using the username and password** from tenant config"

How does the highlighted part relate to what is actually tested?

Reply:
> How does the highlighted part relate to what is actually tested?

Highlted part is related because we are providing the credientials like this: 
```php
$tenant = Tenant::create([
        'tenancy_db_name' => $name,
        'tenancy_db_username' => 'root',
        'tenancy_db_password' => 'password',
    ]);
```

And when we do assert that the database is created, that means credentials in tenant config are being used to create the database successfully. So "db can be created **by using the username and password** from tenant config". I don't know why this is confusing.

Reply:
Resolved on BC.